src/app/api/v1/user/verify-email/route.ts

```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { NextRequest, NextResponse } from "next/server";
+import { verifyEmailHash } from "../../../../../db/tables/emailAddresses.js";
+export async function GET(req: NextRequest) {
+  try {
+    const query = req.nextUrl.searchParams;
+    const token = query.get("token");
+    await verifyEmailHash(token);
+    return NextResponse.redirect(
+      process.env.SERVER_URL + "/user/settings",
+      301
+    );
+  } catch (e) {
+    console.error(e);
+    return NextResponse.json({ success: false }, { status: 500 });
```

Comment:
Q: Does this need a `message`?

Reply:
I don't think so, since we don't know what the cause of this error is - I don't think we want to expose internal eror messages to the outside world.